casper/src/main/rholang/MakeMint.rho

```diff
@@ -30,7 +56,17 @@ in {
         contract internalMakePurse(balance, return) = {
           //balance must be a name that NonNegativeNumber contract is listening on
           new thisPurse in {
-            contract thisPurse(@=*decr, @amount, success) = { balance!("sub", amount, *success) } |
+            new letCh in { // let thisDecr = ...
+              letCh!(bundle0{*thisPurse|*decr}) | for (thisDecr <- letCh) {
+                contract thisPurse(@=*thisDecr, @amount, return) = {
+                  new success in {
+                    balance!("sub", amount, *success) | for (@subOk <- success) {
+                      return!(subOk, *decr)
```

Comment:
Is returning `*decr` like signing the message?

Reply:
yes; it's proving that you were created by the right mint.

Note that we don't hand it out to just anybody; only to those who had the mint to construct `bundle0{*thisPurse|*decr}`.

and just using `decr` there wouldn't work, because an attacker could call "deposit' with a fake purse and we'd hand `decr` to the fake purse. So what we hand out is specific to `src`.

Reply:
yes, like a hash.